plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -0,0 +1,293 @@
+  | "TASK_UNKNOWN"
+  | "TASK_UNREACHABLE";
+
+export interface Docker {
+  forcePullImage: boolean;
+  image: string;
+}
+
+export interface Label {
+  key: string;
+  value: string;
+}
+
+export interface JobHistorySummary {
+  failureCount: number;
+  lastFailureAt: string | null;
+  lastSuccessAt: string | null;
+  successCount: number;
+}
+
+export interface ScheduleConnection {
+  nodes: Schedule[];
+}
+
+export interface Schedule {
+  cron: string;
+  enabled: boolean;
+  id: string;
+  startingDeadlineSeconds: number;
+  timezone: string;
+}
+
+export type JobStatus =
+  | "ACTIVE"
+  | "COMPLETED"
+  | "FAILED"
+  | "INITIAL"
+  | "RUNNING"
+  | "SCHEDULED"
+  | "STARTING"
+  | "UNSCHEDULED";
+
+export const typeDefs = `
+type Job {
+  activeRuns: JobRunConnection!
+  command: String!
+  cpus: Float!
+  description: String!
+  disk: Float!
+  docker: Docker
+  id: ID!
+  jobRuns: JobRunConnection!
+  json: String!
+  labels: [Label]!
+  lastRunsSummary: JobHistorySummary!
+  lastRunStatus: JobRunStatusSummary!
+  mem: Int!
+  name: String!
+  schedules: ScheduleConnection!
+  scheduleStatus: JobStatus!
+}
+type JobRunConnection {
+  longestRunningActiveRun: JobRun
+  nodes: [JobRun]!
+}
+type JobRun {
+  dateCreated: Int!
+  dateFinished: Int!
+  jobID: String!
+  status: JobRunStatusSummary!
+  tasks: JobTaskConnection!
+}
+type JobTaskConnection {
+  longestRunningTask: JobTask!
+  nodes: [JobTask]!
+}
+type JobTask {
+  dateCompleted: Int!
+  dateStarted: Int!
+  status: JobTaskStatus!
+  taskId: String!
+}
+type JobHistorySummary {
+  failureCount: Int!
+  lastFailureAt: String
+  lastSuccessAt: String
+  successCount: Int!
+}
+type Docker {
+  forcePullImage: Boolean!
+  image: String!
+}
+type Label {
+  key: String!
+  value: String!
+}
+type JobRunStatusSummary {
+  status: JobRunStatus
+  time: Int
+}
+type ScheduleConnection {
+  nodes: [Schedule]!
+}
+type Schedule {
+  cron: String!
+  enabled: Boolean!
+  id: ID!
+  startingDeadlineSeconds: Int!
+  timezone: String!
+}
+enum JobTaskStatus {
+  TASK_CREATED
+  TASK_DROPPED
+  TASK_ERROR
+  TASK_FAILED
+  TASK_FINISHED
+  TASK_GONE
+  TASK_GONE_BY_OPERATOR
+  TASK_KILLED
+  TASK_KILLING
+  TASK_LOST
+  TASK_RUNNING
+  TASK_STAGING
+  TASK_STARTED
+  TASK_STARTING
+  TASK_UNKNOWN
+  TASK_UNREACHABLE
+}
+enum JobRunStatus {
+  FAILED
+  NOT_AVAILABLE
+  SUCCESS
+}
+enum JobStatus {
+  ACTIVE
+  COMPLETED
+  FAILED
+  INITIAL
+  RUNNING
+  SCHEDULED
+  STARTING
+  UNSCHEDULED
+}
+type JobConnection {
+  filteredCount: Int!
+  totalCount: Int!
+  nodes: [Job]!
+}
+enum SortOption {
+  ID
+  STATUS
+  LAST_RUN
+}
+enum SortDirection {
+  ASC
+  DESC
+}
+type Query {
+  jobs(
+    filter: String
+    sortBy: SortOption
+    sortDirection: SortDirection
+  ): JobConnection
+  job(
+    id: ID!
+    filter: String
+    sortBy: SortOption
+    sortDirection: SortDirection
+  ): Job
+}
+`;
+
+// Pass implementation of MetronomeClient to the resolvers for easier testability
```

Comment:
Can you explain the problem to me, I don't really see it with the current implementation. That being said I am not married to it, I can also switch it out and mock the respective requests 🤷‍♂️

Reply:
I think the current implementation should just use MetronomeClient directly. It is simpler and it makes easy to add the final solution we want later.

This is not going to stay like this. There will be a way to inject MetronomeClient using Dependency Injection. So the problem with this implementation is that it is trying to solve a problem on a suboptimal way by allowing the resolvers to be configured by passing the client to them, on a scenario where we already know we wanna solve that with the Extension Kid.

Saying more on that topic, if we were not planning to use the IOC container, I would propose we do it like Prisma does, passing the MetronomeClient to the resolvers via the context. That is also the way we currently do on the reference implementation (see repositoryModel.js). That said, that context solution is also suboptimal in the light of the Extension Kid.

Reply:
@Fabs it's gonna take a while for Extension Kid to be fully there, so don't count on it. Some required tasks are in the next sprint.

Reply:
@nLight I ack what you said and I think this is not a problem. For this PR, I think we should not try to do any magic here, just use the MetronomeClient directly and then this week start discussing the timeline to use the Extension Kid. If we can't use the Extension Kid immediately, that is ok.

Reply:
I think you might misunderstand my intentions here. The schema is the default export and we won't expose a way to configure the MetronomeClient or any other implementation there, this should be the work of Extension Kid, I agree.

We use the named export for the resolvers only for easier testing and in this context, I would like to be able to pass in the functions the resolver needs. I think this is nicer than mocking the entire endpoint for each test and being dependant on path changes and stuff like that.

Reply:
I agree that is a better design for testing. I am actually going to try that in another PR I a having problems with.

I just did not want to be understood as the design for the resolvers, because we wanna use extensions.